plugincontainer/compatibility_matrix_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package plugincontainer_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-secure-stdlib/plugincontainer"
+)
+
+const (
+	engineDocker = "docker"
+	enginePodman = "podman"
+	runtimeRunc  = "runc"
+	runtimeRunsc = "runsc"
+)
+
+type matrixInput struct {
+	containerEngine  string
+	containerRuntime string
+	rootlessEngine   bool
+	rootlessUser     bool
+	mlock            bool
+}
+
+func (m matrixInput) String() string {
+	var s string
+	if m.rootlessEngine {
+		s = "rootless "
+	}
+	s += m.containerEngine
+	// Podman does not support configuring the runtime from the SDK.
+	if m.containerEngine != enginePodman {
+		s += ":" + m.containerRuntime
+	}
+	if m.rootlessUser {
+		s += ":" + "nonroot"
+	}
+	if m.mlock {
+		s += ":" + "mlock"
+	}
+	return s
+}
+
+func TestCompatibilityMatrix(t *testing.T) {
+	runCmd(t, "go", "build", "-o=examples/container/go-plugin-counter", "./examples/container/plugin-counter")
```

Comment:
Should we skip the tests on macOS? Currently I get a failure for these with something like `Did not find expected socket file: stat /run/user/1000/docker.sock: no such file or directory`